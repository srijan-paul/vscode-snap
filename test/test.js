let a = 1;
let b = 2;

for (const k of [1, 2, 3]) {
  console.log(k)
}

const mytable = {
  a: 1,
  b: 2
}

let t = mytable.property

function myfn() {
  return false;
}


let c = myfn()
