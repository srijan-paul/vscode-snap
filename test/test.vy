const math = import('math')
const io   = import('io')

const min = 1
const max = 100

const num = math.random(min, max)

let guess = -1

-- Repeatedly take input from user
-- until the guess is correct.

do {
  guess = io.get_int('Enter a guess: ')

  if (guess < num) print("Too low!")
  else (guess > num) print ("Too high!")
} while(num != guess)


print("Good job! The number was indeed ", num)

-- 
let a = "aaa"
let b = "bbb"
let c = a..b

--
let x = 10
let dx = 1

x += dx


--

const vector = {
  make(x, y) {
    return setproto( { x: x, y: y }, self)
  },

  log() {
    print(self.x, self.y)
  },
  
  foo: 'bar'
}

const o = vector:make(0, 0)
o:log()