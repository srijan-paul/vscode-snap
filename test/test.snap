const lib = import "my-lib"

fn double(k) {
  return k * 2
}

const four = 4
const eight = double(four)

if (eight != 8) error("Wow this language doesn't work!")

const myTable = {
  key: "value",
  another_key: 123
}

class Point {
  init(x, y) {
    @x = x
    @y = y
  }
}

return myTable.key